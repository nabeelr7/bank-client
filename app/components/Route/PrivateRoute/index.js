/* eslint-disable react/prop-types */
import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import { createStructuredSelector } from 'reselect';
import { makeSelectIsLogged } from 'containers/App/selectors';
import { useSelector } from 'react-redux';
import { routes } from 'utils';

const stateSelector = createStructuredSelector({
  isLogged: makeSelectIsLogged(),
});

export default function PrivateRoute({
  component: Component,
  restricted,
  ...rest
}) {
  const { isLogged } = useSelector(stateSelector);

  return (
    <Route
      {...rest}
      render={(props) =>
        isLogged ? <Component {...props} /> : <Redirect to={routes.home.path} />
      }
    />
  );
}
