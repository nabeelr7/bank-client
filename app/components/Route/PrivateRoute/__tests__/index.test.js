import React from 'react';
import { render } from '@testing-library/react';

import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';

import configureStore from 'configureStore';
import { ConnectedRouter } from 'connected-react-router';
import PrivateRoute from '../index';

describe('<PrivateRoute />', () => {
  const history = createMemoryHistory();
  const store = configureStore({}, history);

  it('should render a PrivateRoute', () => {
    const component = <div />;
    const { container } = render(
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <PrivateRoute component={component} restricted />
        </ConnectedRouter>
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
