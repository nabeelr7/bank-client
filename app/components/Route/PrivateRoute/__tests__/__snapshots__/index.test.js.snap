// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivateRoute /> should render a PrivateRoute 1`] = `null`;
